core/amoebotsystem.h

```diff
@@ -52,40 +53,44 @@ class AmoebotSystem : public System, public RandomNumberGenerator {
   // Returns a reference to the object list.
   virtual const std::deque<Object*>& getObjects() const final;
 
-  // Functions for measuring the progress of the system. numMovements returns
-  // the total number of expansions and contractions performed by the system's
-  // particles. numRounds returns the number of completed asynchronous rounds;
-  // recall that an asynchronous round is complete when each particle has been
-  // activated at least once.
-  unsigned int numMovements() const final;
-  unsigned int numRounds() const final;
-
-  // Inserts a particle or a object, respectively, into the system. A particle can be contracted or
-  // expanded. Fails if the respective node(s) are already occupied.
+  // Inserts a particle or an object, respectively, into the system. A particle
+  // can be contracted or expanded. Fails if the respective node(s) are already
+  // occupied.
   void insert(AmoebotParticle* particle);
   void insert(Object* object);
 
-  // Functions for logging the progress of the system. registerMovement
-  // increments the total number of movements the system has made by the given
-  // amount. registerActivation logs that the given particle has been activated;
-  // when all particles have been activated at least once, it resets its logging
-  // and increments the number of completed asynchronous rounds by one.
-  // ISSUE: registerActivation simply inserts particles into a collection and
-  // increments the rounds when the collection size equals the system size,
-  // which relies on the assumption that #activations per round = #particles.
-  // See activate.
-  void registerMovement(unsigned int num = 1);
+  // Functions for logging system progress. registerMovement logs the given
+  // number of movements the system has made. registerActivation logs that the
+  // given particle has been activated. When all particles have been activated
+  // at least once, this resets its logging and triggers registerRound(), which
+  // commits all counts and measures to their histories and increments the
+  // number of completed asynchronous rounds by one.
+  void registerMovement(unsigned int numMoves = 1);
   void registerActivation(AmoebotParticle* particle);
+  void registerRound();
+
+  // Various access functions for metrics (counts and measures). getCounts
+  // (resp., getMeasures) returns a reference to the count (resp., measure)
+  // list. getCount (resp., getMeasure) return a reference to the named count
```

Comment:
Small change: `returns` instead of `return`

Reply:
I've addressed this in a new commit.